projects/storefrontapp-e2e-cypress/cypress/integration/b2b/order-approval/b2b-order-approval.e2e-spec.ts

```diff
@@ -0,0 +1,234 @@
+    orderApproval.visitOrderApprovalDetailPage();
+
+    assertButtons();
+    assertPermissionResults(sampleData.approvalOrderDetail.order);
+    assertOrderDetails();
+  });
+
+  it('should approve the order', () => {
+    orderApproval.loginB2bApprover();
+    orderApproval.getOrderApprovalDetail();
+    orderApproval.visitOrderApprovalDetailPage();
+    cy.wait('@orderApprovalPending');
+    orderApproval.makeDecision();
+    orderApproval.getApprovedOrderApprovalDetail();
+
+    cy.get('cx-order-approval-detail-form .btn-primary').eq(1).click();
+    cy.get('cx-order-approval-detail-form textarea').type('test approval');
+    cy.get('cx-order-approval-detail-form .btn-primary').click();
+    cy.wait('@orderApprovalApproved');
+
+    cy.get('cx-order-approval-detail-form').should('contain', 'Back To List');
+    cy.get('cx-order-approval-detail-form .btn-primary').should('not.exist');
+    assertPermissionResults(sampleData.approvedOrderDetails.order);
+  });
+
+  it('should reject the order', () => {
+    orderApproval.loginB2bApprover();
+    orderApproval.getOrderApprovalDetail();
+    orderApproval.visitOrderApprovalDetailPage();
+    cy.wait('@orderApprovalPending');
+
+    orderApproval.makeDecision();
+    orderApproval.getRejectedOrderApprovalDetail();
+
+    cy.get('cx-order-approval-detail-form .btn-primary').eq(0).click();
+    cy.get('cx-order-approval-detail-form textarea').type('test rejection');
+    cy.get('cx-order-approval-detail-form .btn-primary').click();
+    cy.wait('@orderApprovalRejected');
+
+    cy.get('cx-order-approval-detail-form').should('contain', 'Back To List');
+    cy.get('cx-order-approval-detail-form .btn-primary').should('not.exist');
+    assertPermissionResults(sampleData.rejectedOrderDetails.order);
+  });
+});
+
+function assertButtons() {
+  cy.get('cx-order-approval-detail-form a.btn').should(
+    'contain',
+    'Back To List'
+  );
+  cy.get('cx-order-approval-detail-form .btn-primary')
+    .eq(0)
+    .should('contain', 'Reject Order...');
+  cy.get('cx-order-approval-detail-form .btn-primary')
+    .eq(1)
+    .should('contain', 'Approve Order...');
+}
+
+function assertPermissionResults(order) {
+  cy.get('cx-order-details-approval-details').within(() => {
+    cy.get('tr').should('have.length', order.permissionResults.length);
+  });
+
+  order.permissionResults.forEach((permission, index) => {
+    cy.get('cx-order-details-approval-details tr')
+      .eq(index)
+      .within(() => {
+        cy.get('.cx-approval-approverName').should(
+          'contain',
+          permission.approverName
+        );
+      });
+  });
+}
+
+function assertOrderDetails() {
+  // assert order status
+  cy.get('cx-order-details-items .cx-list-status').should(
+    'contain',
+    sampleData.statusPendingApproval
+  );
+
+  // assert products
+  cy.get('.cx-item-list-row .cx-link').should(
+    'contain',
+    sampleData.approvalOrderDetail.order.entries[0].product.name
```

Comment:
Can we shorten this to
firstOrderEntry = sampleData.approvalOrderDetail.order.entries[0]
and then assert on firstOrderEntry.* ?
(law of Demeter)